addons/intel/e830.go

```diff
@@ -35,64 +40,65 @@ type E830PluginData struct {
 
 // OnPTPConfigChangeE830 is called on PTP config change for e830 plugin
 func OnPTPConfigChangeE830(_ *interface{}, nodeProfile *ptpv1.PtpProfile) error {
-	glog.Info("calling onPTPConfigChange for e830 plugin")
+	glog.Infof("calling onPTPConfigChange for e830 plugin (%s)", *nodeProfile.Name)
 	var opts E830Opts
 	var err error
 	var optsByteArray []byte
+
+	if (*nodeProfile).PtpSettings == nil {
+		(*nodeProfile).PtpSettings = make(map[string]string)
+	}
+
 	for name, raw := range (*nodeProfile).Plugins {
 		if name == pluginNameE830 {
+			// Parse user-specified config
 			optsByteArray, _ = json.Marshal(raw)
 			err = json.Unmarshal(optsByteArray, &opts)
 			if err != nil {
 				glog.Error("e830 failed to unmarshal opts: " + err.Error())
 			}
-			if (*nodeProfile).PtpSettings == nil {
-				(*nodeProfile).PtpSettings = make(map[string]string)
-			}
+
+			allDevices := opts.allDevices()
+			glog.Infof("Initializing e830 plugin for profile %s and devices %v", *nodeProfile.Name, allDevices)
+
+			// Setup clockID (prefer ice modue clock ID for e830)
 			iceClockID, iceErr := getClockIDByModule("ice")
```

Comment:
In this case we shouldn't be doing that. The problem is that in the future there will be several "ice" driver reported DPLLs, some of them even unrelated to network silicone. For example, infamous TS_PLL (the one that generates clock for the GNR-D timestamping units) will present itself through the netlink DPLL API.
Here we should derive the clock ID through the PCI.